src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -1299,6 +1299,18 @@ public static double Round(double value, int digits)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static double Round(double value, MidpointRounding mode)
         {
+            // Inline single-instruction modes
```

Comment:
> `Round(value)` isn't exactly "single-instruction" :)

why not? 
![image](https://user-images.githubusercontent.com/523221/150364309-f7c53e70-328f-47a3-b640-d3fc673defd4.png)

Reply:
Will the [Intrinsic] implementation of `Round(value)` be used in all cases?

Reply:
> Will the [Intrinsic] implementation of `Round(value)` be used in all cases?

For all platforms except ARM32 (but my change still improves codegen for it) - do you think it worth leaving a note for arm32?

Reply:
Mono also treats Round as an intrinsic on all of its execution modes: interp, jit, llvm

Reply:
I was mostly being cute.  What you have is fine.

Reply:
> And if "yes", why does the method need an ~50 line C# implementation?

didn't notice this part
That is a good question and my understanding that it's still used in tier0, however, it doesn't make much sense to me as we're going to spend a lot of time jitting that method for tier0. Also it might help some new runtimes to ramp up? or platforms where there are no special instructions for that.

Reply:
> Will the [Intrinsic] implementation of Round(value) be used in all cases? (And if "yes", why does the method need an ~50 line C# implementation?)

We need a correct software implementation for new platforms and for indirect invocation (e.g. reflection, delegates, function pointers, etc). It's also needed on x86/x64 if you're running on hardware from before ~2007.